File: Java-Module/src/main/java/com/sim/javamodule/domain/product/service/dto/ProductSimpleInfoResponse.java
package com.sim.javamodule.domain.product.service.dto;

import com.sim.javamodule.domain.product.domain.Product;
import lombok.Getter;

public record ProductSimpleInfoResponse(Long productId, String name, Long cost) {
    public static ProductSimpleInfoResponse mapProductToProductSimpleInfoResponse(final Product product){
        return new ProductSimpleInfoResponse(product.getId(), product.getName(), product.getCost());
    }
}

File: Kotlin-Module/src/main/kotlin/com/sim/kotlin/domain/product/service/dto/ProductSimpleInfoResponse.kt
package com.sim.kotlin.domain.product.service.dto

import com.sim.kotlin.domain.product.domain.Product

data class ProductSimpleInfoResponse(
    val productId: Long?,
    val name: String,
    val cost: Long,
) {
    companion object{
        fun mapProductToProductSimpleInfoResponse(product: Product) : ProductSimpleInfoResponse{
            return ProductSimpleInfoResponse(product.id, product.name, product.cost)
        }
    }
}